### ISSUE TITLE
View extension doesn't work with partials when using handlebars

### ISSUE THREAD

**Issue by felixmc** [balderdashy/sails#2847]
I am using Sails v0.11 and express-handlebars v2.0.0.

I have the following in my `config/views.js`:

``` javascript
module.exports.views = {
	engine:    'handlebars',
	extension: 'hbs',
	partials:  'partials/',
	layout:    'layout'
};
```

As you can see I am just setting the view engine to handlebars and the view extension to `.hbs` (as opposed to the default `.handlebars`) and setting the partials directory and default layout.

My `views/` directory looks something like this:
```
views
  \-- layout.hbs
  \-- home.hbs        
  \-- post.hbs
  \-- partials
        \-- login.hbs
        \-- post.hbs
        \-- new.hbs
```

In my `layout.hbs` I try to load the "new" partial using handlebars syntax:
``` handlebars
{{> new}}
```

But when I open up a page in the browser I see the following error:
``` json
{
   "message": "The partial new could not be found",
   "name": "Error"
}
```

And the following stack trace in the console:
``` json
error: Sending 500 ("Server Error") response:
 Error: The partial new could not be found
    at new Error (<anonymous>)
    at Error.Exception (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/exception.js:13:41)
    at Object.invokePartial (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:179:11)
    at Object.invokePartialWrapper [as invokePartial] (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:46:39)
    at Object.eval (eval at <anonymous> (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:171:23), <anonymous>:6:17)
    at ret (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:137:30)
    at ret (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:422:21)
    at ExpressHandlebars._renderTemplate (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/lib/express-handlebars.js:296:12)
    at ExpressHandlebars.<anonymous> (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/lib/express-handlebars.js:170:21)
    at /opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/promise/core.js:33:15
    at flush (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/promise/node_modules/asap/asap.js:27:13)
    at process._tickDomainCallback (node.js:492:13) { [Error: The partial new could not be found]
  description: undefined,
  fileName: undefined,
  lineNumber: undefined,
  message: 'The partial new could not be found',
  name: 'Error',
  number: undefined }
warn: res.serverError() :: When attempting to render error page view, an error occured (sending JSON instead).  Details:  Error: The partial new could not be found
    at new Error (<anonymous>)
    at Error.Exception (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/exception.js:13:41)
    at Object.invokePartial (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:179:11)
    at Object.invokePartialWrapper [as invokePartial] (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:46:39)
    at Object.eval (eval at <anonymous> (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/javascript-compiler.js:171:23), <anonymous>:6:17)
    at ret (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/runtime.js:137:30)
    at ret (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/handlebars/dist/cjs/handlebars/compiler/compiler.js:422:21)
    at ExpressHandlebars._renderTemplate (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/lib/express-handlebars.js:296:12)
    at ExpressHandlebars.<anonymous> (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/lib/express-handlebars.js:170:21)
    at /opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/promise/core.js:33:15
    at flush (/opt/apps/stiikr/node_modules/sails/node_modules/express-handlebars/node_modules/promise/node_modules/asap/asap.js:27:13)
    at process._tickDomainCallback (node.js:492:13) { [Error: The partial new could not be found]
  description: undefined,
  fileName: undefined,
  lineNumber: undefined,
  message: 'The partial new could not be found',
  name: 'Error',
  number: undefined }
```

Basically sails and/or express-handlebars are having trouble locating the `new` partial even though there is a file at `views/partials/new.hbs`.

However if I were to rename `new.hbs` to `new.handlebars`, sails is able to find and render the partial just fine. This means that the `extension` property in `config/views.js` doesn't seem to apply to partials, only to layouts and regular views, at least in the case of handlebars.

**Comment by sgress454** [comment]
Hi @felixmc, thanks for the report.  It looks like the handlebars configuration wasn't passing through the custom extension setting correctly--I've [patched that](https://github.com/balderdashy/sails/commit/82098f6ab36449ccedb2d3fcab81fe001ff7c857), and it will go out in the next release.  In the meantime, it should work if you name your partial with a `.handlebars` extension.
